sql/catalyst/src/main/java/org/apache/spark/sql/catalog/v2/CatalogV2Util.scala

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalog.v2
+
+import java.util
+import java.util.Collections
+
+import scala.collection.JavaConverters._
+
+import org.apache.spark.sql.catalog.v2.TableChange.{AddColumn, DeleteColumn, RemoveProperty, RenameColumn, SetProperty, UpdateColumnComment, UpdateColumnType}
+import org.apache.spark.sql.types.{StructField, StructType}
+
+private[sql] object CatalogV2Util {
```

Comment:
This has a good intention, but there was an argument, too. According to `SPARK-16813 Remove private[sql] and private[spark] from catalyst package`, shall we not to use `private[sql]`?
cc @HyukjinKwon

Reply:
Sounds reasonable to me. Where should I move this to so that it doesn't show up in the API?

Reply:
For that, shall we create `o.a.s.sql.catalog.v2.util` package and add them to `ignoreUndocumentedPackages` of `SparkBuild.scala`?

BTW, this is a simple move from `object TestTableCatalog` (in `test` source)? Could you make a separate PR for that, please?

Reply:
In the past, I've been told that PRs that add utilities without using them are discouraged. Is this a different case? I'm happy to do it either here or in a different PR, but this conflicts with what I've been told by other committers.

Reply:
First of all, refactoring from `TestTableCatalog` to `CatalogV2Util` is not the one you mentioned. I believe this one should be done before this.

For `transformer`, yes, I agree. It's on the edge. But, Apache Spark community also does the similar thing to split big PRs.

Reply:
Since I understand your concerns, cc @cloud-fan , @gatorsmile , @jiangxb1987 , @rxin .

Reply:
Opened https://github.com/apache/spark/pull/24813